### PR TITLE
PAT-299: Store all banding information in the Pathfinder DB

### PR DESCRIPTION
**Pull request by timharrison-moj**
This PR records the actual, recommended and minimum bands into the Pathfinder database when the recommended band is accepted or is overridden. This will be useful for reporting when the bands have been altered. The form contains any reason for override.

### CODE REVIEW

File: server/routes/createCim.js [hunk 1]
```diff
@@ -48,8 +48,12 @@ module.exports = function PathfinderCimRoutes({ offenderService, pathfinderServi
     const bandingResult = await pathfinderService.calculateBanding(nominalId, formData)
     const recommendedBandWarning = `Risk band ${bandingResult.recommendedBand}`
     const { recommendedRiskBand = {} } = formData
+
     recommendedRiskBand.recommendedBandWarning = recommendedBandWarning
     recommendedRiskBand.band = bandingResult.recommendedBand
+    recommendedRiskBand.autoBand = bandingResult.recommendedBand
```
Comment (yasinmustafa) [inline server/routes/createCim.js]: This line seems to create a duplicate entry of the data in the JSON.
The JSON structure looked similar to the following:
`recommendedRiskBand: {band: 1, acceptReccomendation: 'NO', furtherDetails: 'Want to override'}`
Don't think we need to change it to duplicate an attribute (that isn't used):
`recommendedRiskBand: {band: 1, autoBand: 1, acceptReccomendation: 'NO', furtherDetails: 'Want to override'}`

The purpose of the recommendedRiskBand part in the JSON is to hold what the algorithm recommends and displays that info in a cim task (the fields are linked to config)

File: server/routes/createCim.js [hunk 2]
```diff
@@ -79,9 +83,21 @@ module.exports = function PathfinderCimRoutes({ offenderService, pathfinderServi
 
   async function updateBandIfFinalised(sectionConfig, payloadFields, fields, nominalId, updatedPayload, allErrors) {
     if (saveBandOnSubmit(sectionConfig, payloadFields, fields)) {
-      await pathfinderService.updateBand(nominalId, updatedPayload.recommendedRiskBand.band)
+      // Accept the banding recommendation
+      await pathfinderService.updateBands(
+        nominalId,
+        updatedPayload.recommendedRiskBand.band,
+        updatedPayload.recommendedRiskBand.autoBand,
```
Comment (yasinmustafa) [inline server/routes/createCim.js]: offshoot of my previous comment
`recommendedRiskBand.autoBand` is a duplicate of `recommendedRiskBand.band`

so I would remove autoBand and just pass in recommendedRiskBand.band twice.

File: server/routes/createCim.js [hunk 3]
```diff
@@ -79,9 +83,21 @@ module.exports = function PathfinderCimRoutes({ offenderService, pathfinderServi
...
+        updatedPayload.recommendedRiskBand.minimumBand
+      )
     } else if (sectionConfig.overrideBandOnSubmit && isNilOrEmpty(allErrors)) {
-      await pathfinderService.updateBand(nominalId, updatedPayload.overrideRiskBandTask.overrideRiskBand.answer)
+      // Use the override band in place of the recommended band
+      await pathfinderService.updateBands(
+        nominalId,
+        updatedPayload.overrideRiskBandTask.overrideRiskBand.answer,
+        updatedPayload.recommendedRiskBand.autoBand,
```
Comment (yasinmustafa) [inline server/routes/createCim.js]: following on from previous points...
can use `recommendedRiskBand.band` here

File: server/routes/createCim.js (hunk 2 above)
Comment (timharrison-moj) [reply server/routes/createCim.js]: Agreed.

File: server/routes/createCim.js (hunk 3 above)
Comment (timharrison-moj) [reply server/routes/createCim.js]: Agreed.

File: server/routes/createCim.js (hunk 1 above)
Comment (timharrison-moj) [reply server/routes/createCim.js]: I wasn't sure whether the recommendedBand.band was actually overwritten by the overrideBand at the time and only discovered later that a distinct value is used in the override section.  I'll take it out.

**Pull request merged**